src/config.d.ts

```diff
@@ -0,0 +1,27 @@
+// These types be improved once recursive type references are released (https://github.com/microsoft/TypeScript/pull/33050)
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface JSONArray extends Array<JSONValue> {}
+type JSONValue = string | number | boolean | JSONObject | JSONArray
```

Comment:
This feels a bit boilerplaty that we have to define this ourselves. I assume you wanted a type that is `anything BUT null|undefined`.

Is there no typescript-native way to do this?

Reply:
Hm it's not the same since there are many more types that are not valid JSONValues: `Symbol`, `Function`, `Enum`, any class, maybe more.

So it's correct that a valid JSONValue is defined as shown in L5. Also safer bc more defensive.

Reply:
ok, just feels like something as basic as JSON types this should be baked in to typescript. but if there's no better way to do this: 👍